manifests/istio-telemetry/kiali/templates/kiali-operator-clusterrole.yaml

```diff
@@ -0,0 +1,255 @@
+{{- if .Values.kiali.operator.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kiali-operator-{{ .Release.Namespace }}
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+rules:
+- apiGroups: [""]
+  resources:
+  - configmaps
+  - endpoints
+  - events
+  - persistentvolumeclaims
+  - pods
+  - serviceaccounts
+  - services
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups: [""]
+  resources:
+  - namespaces
+  verbs:
+  - get
+  - list
+  - patch
+- apiGroups: ["apps"]
+  resources:
+  - deployments
+  - replicasets
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups: ["monitoring.coreos.com"]
+  resources:
+  - servicemonitors
+  verbs:
+  - create
+  - get
+- apiGroups: ["apps"]
+  resourceNames:
+  - kiali-operator
+  resources:
+  - deployments/finalizers
+  verbs:
+  - update
+- apiGroups: ["kiali.io"]
+  resources:
+  - '*'
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups: ["rbac.authorization.k8s.io"]
+  resources:
+  - clusterrolebindings
+  - clusterroles
+  - rolebindings
+  - roles
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups: ["apiextensions.k8s.io"]
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups: ["extensions"]
+  resources:
+  - ingresses
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups: ["route.openshift.io"]
```

Comment:
Can you confirm that adding these api groups works for upstream K8s 1.17?

Reply:
These fine for upstream K8s. We have people using this same yaml in upstream Kiali on upstream k8s and it works. The way k8s does this is it still adds these API group permissions but if they do not exist, it is just an no-op. However, these are required to work properly when people install on OpenShift.

Without something like this, OpenShift users are required (as they are today) to perform additional steps after istioctl installs Istio in order to get Kiali to work properly. As it is currently, istioctl users wanting to install Kiali on OpenShift need to perform additional commands after istioctl installs everything - see https://istio.io/docs/tasks/observability/kiali/#running-on-openshift

Reply:
SGTM.